cardinal/gamestate/ecb.go

```diff
@@ -24,7 +24,7 @@ type EntityCommandBuffer struct {
 
 	compValues         PrimitiveStorage[compKey]
 	compValuesToDelete PrimitiveStorage[compKey]
-	typeToComponent    map[types.ComponentID]types.ComponentMetadata
+	typeToComponent    PrimitiveStorage[types.ComponentID]
```

Comment:
I disagree with this. We want to keep type information here; we don't want future maintainers to have to know to cast this to the ComponentMetadata type.

The compValuesToDelete field was more ok because it's just a Set and the 'bool' didn't really matter. The compValues field was more ok because there actually is a mix of types that will be stored in the values, but this field has a specific job with a single type.

Reply:
The whole reason for these series of PRs was for replacing ALL maps with the Storage interface. Just to be clear about this. So it's not about whether your ok with one and not ok with the other, the approval should also include a wholistic opinion here, whether you approve ALL maps to be replaced with storage. 

Anyway this was more of a tentative PR, scott discussed a possibility of using a PrimitiveStore[K, V] rather then PrimitiveStore[K], so ALL of PrimitiveStorage might be completely replaced with a new design and that replacement is actually more inline with what you're talking about here (You did however explicitly disagree with the PrimitiveStorage[K, V] last week so I'm not sure about this). Anyway I'll just Draft this for now and we'll talk more when Scott gets back.